include/deal.II/grid/manifold.h

```diff
@@ -723,6 +753,23 @@ class FlatManifold : public Manifold<dim, spacedim>
   get_new_point(const std::vector<Point<spacedim> > &surrounding_points,
                 const std::vector<double>           &weights) const;
 
+  /**
+   * Compute a new set of points around the given points
+   * @p surrounding_points. @p weights is a table with as many columns as
+   * @p surrounding_points.size(). The number of rows in @p weights determines
+   * how many new points will be computed and appended to the last input
+   * argument @p new_points. After exit of this function, the size of
+   * @p new_points equals the size at entry plus the number of rows in
+   * @p weights.
+   *
+   * For this particular implementation, an interpolation of the
+   * @p surrounding_points according to the @p weights is performed.
```

Comment:
*all* manifolds interpolate. The difference is that the `FlatManifold` simply takes the (weighted) arithmetic mean of the given points in the Cartesian coordinate system in which they are provided.